packages/server-wallet/src/bn.ts

```diff
@@ -0,0 +1,30 @@
+import {BigNumber as EthersBigNumber, BigNumberish} from 'ethers';
+import {Uint256} from './store-types';
+
+type T = BigNumberish;
+const compare = (method: 'eq' | 'lt' | 'gt' | 'lte' | 'gte') => (a: T, b: T): boolean =>
+  typeof a === typeof b && EthersBigNumber.from(a)[method](b);
+
+const op = (method: 'add' | 'sub' | 'mul') => (a: T, b: T) => {
+  if (typeof a === 'string' || typeof b === 'string') {
+    return EthersBigNumber.from(a)
+      [method](b)
+      .toHexString();
+  } else {
+    return EthersBigNumber.from(a)
+      [method](b)
+      .toNumber();
+  }
+};
+
+export const BigNumber = {
```

Comment:
This was partially playing around with the idea of not using `BigNumber` class instances.

I'd remove this, and depend on the types defined in `wallet-core`, with the goal of not adding yet another `State` type.